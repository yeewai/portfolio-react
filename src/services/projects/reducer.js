import data from 'projectsData.json';

export const defaultState = data;

const projectsReducer = (state = defaultState, action) => {
    switch(action.type) {
        default:
            return state;
    }
}

export default projectsReducer;
