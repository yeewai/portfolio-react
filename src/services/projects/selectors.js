const sortByDateFn = (a,b) => {
  return new Date(b.date) - new Date(a.date);
}

export const getArt = (state) => state.projects.art.sort(sortByDateFn);
export const getCode = (state) => state.projects.code.sort(sortByDateFn);

export const getByName = ( state, params ) => (
    ["art", "code"].includes(params.projectType)
    ? state.projects[params.projectType].find( p => p.slug === params.projectTitle )
    : null
)
