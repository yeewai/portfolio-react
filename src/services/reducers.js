import { combineReducers } from 'redux';

import { routerReducer } from 'react-router-redux';
import projectsReducer from './projects';

const rootReducer = combineReducers({
    router: routerReducer,
    projects: projectsReducer
});

export default rootReducer;
