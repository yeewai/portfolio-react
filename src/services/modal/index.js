import modalReducer from "./reducer";
import * as modalActions from "./actions";
import * as modalSelectors from "./selectors";


export { modalActions, modalSelectors,};

export default modalReducer;
