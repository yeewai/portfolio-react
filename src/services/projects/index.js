import projectsReducer from "./reducer";
import * as projectsSelectors from "./selectors";

export { projectsSelectors };

export default projectsReducer;
