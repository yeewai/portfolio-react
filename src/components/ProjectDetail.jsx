import React from 'react';
import { connect } from 'react-redux';
import StickyBox from "react-sticky-box";

import { projectsSelectors } from 'services/projects';

export const ProjectDetail = ({ project }) => {
    if ( !project ) { return <h2 id="no-match">There is no matching project</h2>; }

    return (
        <article className="projectDetail row justify-content-center">
            <section className="images col-sm-8">
                { project.images.map( (image, i) => (
                    <img className="img-fluid" src={image.url} alt={image.alt} key={`image-${i}`}/>
                )) }
            </section>
            <div className="col-sm">
                <StickyBox className="project-description card">
                    <section className="card-body">
                        <h2 className="card-title">{project.title}</h2>
                        <h4 className="card-subtitle mb-2 text-muted">{project.subtitle}</h4>
                        <ul className="details">
                            { project.github ? <li><a href={project.github} target="_blank">github</a></li> : null }
                            { project.link ? <li><a href={project.link} target="_blank">View in Action</a></li> : null }
                            <li><time>{project.date}</time></li>
                            <li><strong>Tools</strong>: {project.tools}</li>
                        </ul>
                        <section className="description" dangerouslySetInnerHTML={{__html: project.description}} />

                    </section>
                </StickyBox>
            </div>

        </article>
    );
}

export const mapStateToProps = (state, ownProps) => ({
    project: projectsSelectors.getByName(state, ownProps.match.params)
});

export default connect(mapStateToProps)(ProjectDetail);
