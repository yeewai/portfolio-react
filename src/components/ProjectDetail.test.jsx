import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import sinon from 'sinon';

import { ProjectDetail, mapStateToProps } from './ProjectDetail';
import { projectsSelectors } from 'services/projects';

Enzyme.configure({ adapter: new Adapter() });

describe("Project Detail", () => {
    it('renders', () => {
        const project = {
            images: [
                { url: "a", alt: "b" },
                { url: "c", alt: "d" }
            ],
            title: "Eyyyyyy",
            subtitle: "Sub-eyyyyyyy",
            date: "Some date",
            tools: "Sum tools",
            description: "<p>sup sup sup</p>"
        }
        const wrapper = shallow(<ProjectDetail project={project} />);
        expect(wrapper).toMatchSnapshot();
    });

    it ("renders nothing if there is no project found", () => {
        const wrapper = shallow(<ProjectDetail project={null} />);
        expect(wrapper.find("#no-match")).toHaveLength(1);
    })

    it ('mapStateToProps', () => {
        const sandbox = sinon.sandbox.create();
        sandbox.stub(projectsSelectors, 'getByName').returns("project");

        expect(mapStateToProps({}, {match: {params: "a"}}).project).toEqual("project");

        sandbox.restore();
    })
})
