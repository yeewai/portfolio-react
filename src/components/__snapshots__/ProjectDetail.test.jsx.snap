// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Detail renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    project={
        Object {
            "date": "Some date",
            "description": "<p>sup sup sup</p>",
            "images": Array [
              Object {
                "alt": "b",
                "url": "a",
              },
              Object {
                "alt": "d",
                "url": "c",
              },
            ],
            "subtitle": "Sub-eyyyyyyy",
            "title": "Eyyyyyy",
            "tools": "Sum tools",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="images col-sm-8"
>
          <Connect(OpenModalButton)
                    modalProps={
                              Object {
                                        "alt": "b",
                                        "url": "a",
                                      }
                    }
                    modalType="SHOW_IMAGE"
          >
                    <img
                              alt="b"
                              className="img-fluid"
                              src="a"
                    />
          </Connect(OpenModalButton)>
          <Connect(OpenModalButton)
                    modalProps={
                              Object {
                                        "alt": "d",
                                        "url": "c",
                                      }
                    }
                    modalType="SHOW_IMAGE"
          >
                    <img
                              alt="d"
                              className="img-fluid"
                              src="c"
                    />
          </Connect(OpenModalButton)>
</section>,
        <div
          className="col-sm"
>
          <StickyBox
                    className="project-description card"
          >
                    <section
                              className="card-body"
                    >
                              <h2
                                        className="card-title"
                              >
                                        Eyyyyyy
                              </h2>
                              <h4
                                        class="card-subtitle mb-2 text-muted"
                              >
                                        Sub-eyyyyyyy
                              </h4>
                              <p
                                        className="details"
                              >
                                        <time>
                                                  Some date
                                        </time>
                                        <br />
                                        Tools: 
                                        Sum tools
                              </p>
                              <section
                                        className="description"
                                        dangerouslySetInnerHTML={
                                                  Object {
                                                            "__html": "<p>sup sup sup</p>",
                                                          }
                                        }
                              />
                    </section>
          </StickyBox>
</div>,
      ],
      "className": "projectDetail row justify-content-center",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(OpenModalButton)
              modalProps={
                            Object {
                                          "alt": "b",
                                          "url": "a",
                                        }
              }
              modalType="SHOW_IMAGE"
>
              <img
                            alt="b"
                            className="img-fluid"
                            src="a"
              />
</Connect(OpenModalButton)>,
            <Connect(OpenModalButton)
              modalProps={
                            Object {
                                          "alt": "d",
                                          "url": "c",
                                        }
              }
              modalType="SHOW_IMAGE"
>
              <img
                            alt="d"
                            className="img-fluid"
                            src="c"
              />
</Connect(OpenModalButton)>,
          ],
          "className": "images col-sm-8",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "images-0",
            "nodeType": "class",
            "props": Object {
              "children": <img
                alt="b"
                className="img-fluid"
                src="a"
/>,
              "modalProps": Object {
                "alt": "b",
                "url": "a",
              },
              "modalType": "SHOW_IMAGE",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "b",
                "className": "img-fluid",
                "src": "a",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "images-1",
            "nodeType": "class",
            "props": Object {
              "children": <img
                alt="d"
                className="img-fluid"
                src="c"
/>,
              "modalProps": Object {
                "alt": "d",
                "url": "c",
              },
              "modalType": "SHOW_IMAGE",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "d",
                "className": "img-fluid",
                "src": "c",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <StickyBox
            className="project-description card"
>
            <section
                        className="card-body"
            >
                        <h2
                                    className="card-title"
                        >
                                    Eyyyyyy
                        </h2>
                        <h4
                                    class="card-subtitle mb-2 text-muted"
                        >
                                    Sub-eyyyyyyy
                        </h4>
                        <p
                                    className="details"
                        >
                                    <time>
                                                Some date
                                    </time>
                                    <br />
                                    Tools: 
                                    Sum tools
                        </p>
                        <section
                                    className="description"
                                    dangerouslySetInnerHTML={
                                                Object {
                                                            "__html": "<p>sup sup sup</p>",
                                                          }
                                    }
                        />
            </section>
</StickyBox>,
          "className": "col-sm",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <section
              className="card-body"
>
              <h2
                            className="card-title"
              >
                            Eyyyyyy
              </h2>
              <h4
                            class="card-subtitle mb-2 text-muted"
              >
                            Sub-eyyyyyyy
              </h4>
              <p
                            className="details"
              >
                            <time>
                                          Some date
                            </time>
                            <br />
                            Tools: 
                            Sum tools
              </p>
              <section
                            className="description"
                            dangerouslySetInnerHTML={
                                          Object {
                                                        "__html": "<p>sup sup sup</p>",
                                                      }
                            }
              />
</section>,
            "className": "project-description card",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2
                  className="card-title"
>
                  Eyyyyyy
</h2>,
                <h4
                  class="card-subtitle mb-2 text-muted"
>
                  Sub-eyyyyyyy
</h4>,
                <p
                  className="details"
>
                  <time>
                                    Some date
                  </time>
                  <br />
                  Tools: 
                  Sum tools
</p>,
                <section
                  className="description"
                  dangerouslySetInnerHTML={
                                    Object {
                                                      "__html": "<p>sup sup sup</p>",
                                                    }
                  }
/>,
              ],
              "className": "card-body",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Eyyyyyy",
                  "className": "card-title",
                },
                "ref": null,
                "rendered": "Eyyyyyy",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sub-eyyyyyyy",
                  "class": "card-subtitle mb-2 text-muted",
                },
                "ref": null,
                "rendered": "Sub-eyyyyyyy",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <time>
                      Some date
</time>,
                    <br />,
                    "Tools: ",
                    "Sum tools",
                  ],
                  "className": "details",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Some date",
                    },
                    "ref": null,
                    "rendered": "Some date",
                    "type": "time",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  "Tools: ",
                  "Sum tools",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "description",
                  "dangerouslySetInnerHTML": Object {
                    "__html": "<p>sup sup sup</p>",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "section",
              },
            ],
            "type": "section",
          },
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="images col-sm-8"
>
            <Connect(OpenModalButton)
                        modalProps={
                                    Object {
                                                "alt": "b",
                                                "url": "a",
                                              }
                        }
                        modalType="SHOW_IMAGE"
            >
                        <img
                                    alt="b"
                                    className="img-fluid"
                                    src="a"
                        />
            </Connect(OpenModalButton)>
            <Connect(OpenModalButton)
                        modalProps={
                                    Object {
                                                "alt": "d",
                                                "url": "c",
                                              }
                        }
                        modalType="SHOW_IMAGE"
            >
                        <img
                                    alt="d"
                                    className="img-fluid"
                                    src="c"
                        />
            </Connect(OpenModalButton)>
</section>,
          <div
            className="col-sm"
>
            <StickyBox
                        className="project-description card"
            >
                        <section
                                    className="card-body"
                        >
                                    <h2
                                                className="card-title"
                                    >
                                                Eyyyyyy
                                    </h2>
                                    <h4
                                                class="card-subtitle mb-2 text-muted"
                                    >
                                                Sub-eyyyyyyy
                                    </h4>
                                    <p
                                                className="details"
                                    >
                                                <time>
                                                            Some date
                                                </time>
                                                <br />
                                                Tools: 
                                                Sum tools
                                    </p>
                                    <section
                                                className="description"
                                                dangerouslySetInnerHTML={
                                                            Object {
                                                                        "__html": "<p>sup sup sup</p>",
                                                                      }
                                                }
                                    />
                        </section>
            </StickyBox>
</div>,
        ],
        "className": "projectDetail row justify-content-center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(OpenModalButton)
                modalProps={
                                Object {
                                                "alt": "b",
                                                "url": "a",
                                              }
                }
                modalType="SHOW_IMAGE"
>
                <img
                                alt="b"
                                className="img-fluid"
                                src="a"
                />
</Connect(OpenModalButton)>,
              <Connect(OpenModalButton)
                modalProps={
                                Object {
                                                "alt": "d",
                                                "url": "c",
                                              }
                }
                modalType="SHOW_IMAGE"
>
                <img
                                alt="d"
                                className="img-fluid"
                                src="c"
                />
</Connect(OpenModalButton)>,
            ],
            "className": "images col-sm-8",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "images-0",
              "nodeType": "class",
              "props": Object {
                "children": <img
                  alt="b"
                  className="img-fluid"
                  src="a"
/>,
                "modalProps": Object {
                  "alt": "b",
                  "url": "a",
                },
                "modalType": "SHOW_IMAGE",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "b",
                  "className": "img-fluid",
                  "src": "a",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "images-1",
              "nodeType": "class",
              "props": Object {
                "children": <img
                  alt="d"
                  className="img-fluid"
                  src="c"
/>,
                "modalProps": Object {
                  "alt": "d",
                  "url": "c",
                },
                "modalType": "SHOW_IMAGE",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "d",
                  "className": "img-fluid",
                  "src": "c",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": [Function],
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <StickyBox
              className="project-description card"
>
              <section
                            className="card-body"
              >
                            <h2
                                          className="card-title"
                            >
                                          Eyyyyyy
                            </h2>
                            <h4
                                          class="card-subtitle mb-2 text-muted"
                            >
                                          Sub-eyyyyyyy
                            </h4>
                            <p
                                          className="details"
                            >
                                          <time>
                                                        Some date
                                          </time>
                                          <br />
                                          Tools: 
                                          Sum tools
                            </p>
                            <section
                                          className="description"
                                          dangerouslySetInnerHTML={
                                                        Object {
                                                                      "__html": "<p>sup sup sup</p>",
                                                                    }
                                          }
                            />
              </section>
</StickyBox>,
            "className": "col-sm",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <section
                className="card-body"
>
                <h2
                                className="card-title"
                >
                                Eyyyyyy
                </h2>
                <h4
                                class="card-subtitle mb-2 text-muted"
                >
                                Sub-eyyyyyyy
                </h4>
                <p
                                className="details"
                >
                                <time>
                                                Some date
                                </time>
                                <br />
                                Tools: 
                                Sum tools
                </p>
                <section
                                className="description"
                                dangerouslySetInnerHTML={
                                                Object {
                                                                "__html": "<p>sup sup sup</p>",
                                                              }
                                }
                />
</section>,
              "className": "project-description card",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2
                    className="card-title"
>
                    Eyyyyyy
</h2>,
                  <h4
                    class="card-subtitle mb-2 text-muted"
>
                    Sub-eyyyyyyy
</h4>,
                  <p
                    className="details"
>
                    <time>
                                        Some date
                    </time>
                    <br />
                    Tools: 
                    Sum tools
</p>,
                  <section
                    className="description"
                    dangerouslySetInnerHTML={
                                        Object {
                                                            "__html": "<p>sup sup sup</p>",
                                                          }
                    }
/>,
                ],
                "className": "card-body",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Eyyyyyy",
                    "className": "card-title",
                  },
                  "ref": null,
                  "rendered": "Eyyyyyy",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sub-eyyyyyyy",
                    "class": "card-subtitle mb-2 text-muted",
                  },
                  "ref": null,
                  "rendered": "Sub-eyyyyyyy",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <time>
                        Some date
</time>,
                      <br />,
                      "Tools: ",
                      "Sum tools",
                    ],
                    "className": "details",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Some date",
                      },
                      "ref": null,
                      "rendered": "Some date",
                      "type": "time",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "br",
                    },
                    "Tools: ",
                    "Sum tools",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "description",
                    "dangerouslySetInnerHTML": Object {
                      "__html": "<p>sup sup sup</p>",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "section",
                },
              ],
              "type": "section",
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
